backend/src/apiserver/resource/resource_manager.go

```diff
@@ -179,21 +194,37 @@ func (r *ResourceManager) CreatePipeline(name string, description string, pipeli
 	if err != nil {
 		return nil, util.Wrap(err, "Create pipeline failed")
 	}
+	newPipeline.DefaultVersion.Status = model.PipelineVersionReady
+	err = r.pipelineStore.UpdatePipelineVersionStatus(
+		newPipeline.DefaultVersionId, newPipeline.DefaultVersion.Status)
+	if err != nil {
+		return nil, util.Wrap(err, "Create pipeline version failed")
+	}
```

Comment:
putting two update call to a single transaction.

Reply:
done. but note: the transaction has to be implemented at pipeline_store level in a *single* method, not at the resource manager level. So we'll have a new method in pipeline_store called UpdatedPipelineAndPipelineVersionsStatus, which is only a *temporary* method. Later, CreatePipeline method should switch back to UpdatePipelineStatus, because CreatePipeline should not update version status when the version APIs are there.

Reply:
LGTM